src/xtgeo/grid3d/grid.py

```diff
@@ -388,6 +388,7 @@ def subgrids(self, sgrids):
 
         if sgrids is None:
             self._subgrids = None
+            return
```

Comment:
Should we make a test that resets the subgrids to `None` as well? If I'm not mistaken, that should have failed with `AttributeError: 'NoneType' object has no attribute 'items'` prior to this fix

Reply:
Not sure what you mean here. Would it be better with an empty dict instead of None perhaps?

Reply:
I ment that without the `return` it would not be possible to set `subgrids` to `None` (which is the default value at initialisation). So current implementation has an issue, and starting with a failing test is always good :)

It would've failed earlier than what I said though of course, as it checks two lines below for instance of `OrderedDict`.. 

Otherwise I think it can stay as is. I don't think it would be natural with the following behaviour:
```
>>> grid.subgrids = OrderedDict({"1":[1,2,3,4,5]})
>>> grid.subgrids
OrderedDict([('1', [1, 2, 3, 4, 5])])
>>> grid.subgrids = None
>>> grid.subgrids
{}
```
One could consider changing so that it shouldn't be possible to be anything but a `dict` type (`None` not accepted), but the default values must be changed and it changes behaviour as well. I'd stick with `None` or `dict` as is.